### PR TITLE
add deltalake to eco system doc page

### PR DESCRIPTION
**Pull request by houqp**
- [x] relates to #35017
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -368,6 +368,14 @@ far exceeding the performance of the native ``df.to_sql`` method. Internally, it
 Microsoft's BCP utility, but the complexity is fully abstracted away from the end user.
 Rigorously tested, it is a complete replacement for ``df.to_sql``.
 
+`Deltalake <https://pypi.org/project/deltalake>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/ecosystem.rst:372]: this needs to be the same len as the title

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/ecosystem.rst (hunk 1 above)
Comment (houqp) [reply doc/source/ecosystem.rst]: @jreback updated.

**Review by jreback — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @houqp. CI failures unrelated

**Comment by houqp** [comment]
@mroeschke @jreback I am not seeing this change show up in the official doc, do you know what did I miss in my PR?

**Comment by jreback** [comment]
this will be in 1.3 releasing soon

**Comment by simonjayhawkins** [comment]
you can see the changes at https://pandas.pydata.org/pandas-docs/dev/ecosystem.html?highlight=ecosystem#deltalake

**Comment by houqp** [comment]
thank you @jreback @simonjayhawkins for the context and confirmation!

**Pull request merged**